extra_data/reader.py

```diff
@@ -693,10 +702,11 @@ def is_valid_alias(k, v):
                 alias_dicts.append(alias_def)
             elif isinstance(alias_def, (str, os.PathLike)):
                 # From a file.
+                alias_files.append(alias_def)
```

Comment:
Do we want to normalise the type here?

```suggestion
                alias_files.append(os.fspath(alias_def))
```

This would convert Path objects to strings.

Reply:
I was about to comment, I would prefer it we always have `str` or `Path` internally (as well as externally via the property), and not return combination of types was passed.